patchwork-registries/src/main/java/net/patchworkmc/impl/registries/ExtendedVanillaRegistry.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.patchworkmc.impl.registries;
+
+import net.minecraftforge.registries.ForgeRegistry;
+import net.minecraftforge.registries.IForgeRegistryEntry;
+
+import net.minecraft.util.registry.SimpleRegistry;
+
+// TODO: implement default registry
+public class ExtendedVanillaRegistry<V extends IForgeRegistryEntry<V>> extends SimpleRegistry<V> implements ForgeRegistryProvider {
+	private final ForgeRegistry<V> forgeRegistry;
+	public ExtendedVanillaRegistry(ForgeRegistry<V> forgeRegistry) {
+		this.forgeRegistry = forgeRegistry;
+	}
+
+	public void clear() {
```

Comment:
This should probably actually be implemented as a mixin to SimpleRegistry, and then have a ClearableRegistry interface implemented by the mixin.

Reply:
This class is for forge modded registries, they may need functions that is not supported by SimpleRegistry, like remove(). I'm planning to reimpl this

Reply:
ClearableRegistry is created